src/antidote_db_wrapper.erl

```diff
@@ -0,0 +1,325 @@
+-endif.
+
+%% Gets the most suitable snapshot for Key that has been committed
+%% before CommitTime. If its nothing is found, returns {error, not_found}
+-spec get_snapshot(antidote_db:antidote_db(), key(),
+    snapshot_time()) -> {ok, snapshot(), snapshot_time()} | {error, not_found}.
+get_snapshot(AntidoteDB, Key, CommitTime) ->
+    try
+        antidote_db:fold(AntidoteDB,
+            fun({K, V}, AccIn) ->
+                {Key1, VC, SNAP} = binary_to_term(K),
+                case (Key1 == Key) of %% check same key
+                    true ->
+                        %% check its a snapshot and its time is less than the one required
+                        case (SNAP == snap) and
+                            vectorclock:le(vectorclock:from_list(VC), CommitTime) of
+                            true ->
+                                Snapshot = binary_to_term(V),
+                                throw({break, Snapshot, VC});
+                            _ ->
+                                AccIn
+                        end;
+                    false ->
+                        throw({break})
+                end
+            end,
+            [],
+            [{first_key, term_to_binary({Key})}]),
+        {error, not_found}
+    catch
+        {break, SNAP, VC} ->
+            {ok, SNAP, VC};
+        _ ->
+            {error, not_found}
+    end.
+
+%% Saves the snapshot into AntidoteDB
+-spec put_snapshot(antidote_db:antidote_db(), key(), snapshot_time(),
+    snapshot()) -> ok | error.
+put_snapshot(AntidoteDB, Key, SnapshotTime, Snapshot) ->
+    SnapshotTimeList = vectorclock_to_sorted_list(SnapshotTime),
+    antidote_db:put(AntidoteDB, {binary_to_atom(Key), SnapshotTimeList, snap}, Snapshot).
+
+%% Returns a list of operations that have commit time in the range [VCFrom, VCTo]
+-spec get_ops(antidote_db:antidote_db(), key(), vectorclock(), vectorclock()) -> list().
+get_ops(AntidoteDB, Key, VCFrom, VCTo) ->
+    VCFromDict = vectorclock_to_dict(VCFrom),
+    VCToDict = vectorclock_to_dict(VCTo),
+    try
+        antidote_db:fold(AntidoteDB,
+            fun({K, V}, AccIn) ->
+                {Key1, VC1, OP} = binary_to_term(K),
+                VC1Dict = vectorclock:from_list(VC1),
+                case Key == Key1 of %% check same key
+                    true ->
+                        %% if its greater, continue
+                        case vectorclock:gt(VC1Dict, VCToDict) of
+                            true ->
+                                AccIn;
+                            false ->
+                                %% check its an op and its commit time is in the required range
+                                case not vectorclock:lt(VC1Dict, VCFromDict) of
+                                    true ->
+                                        case (OP == op) of
+                                            true ->
+                                                AccIn ++ [binary_to_term(V)];
```

Comment:
Accumulating at the end of a list is O(N^2). If the list is long, better prepend and swap the list order afterwards.